object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -279,6 +306,276 @@ private static String getExactStringValue(Tree tree, ValueAnnotatedTypeFactory f
     return null;
   }
 
+  @Override
+  public void postAnalyze(ControlFlowGraph cfg) {
+    checkCFG(cfg.getEntryBlock(), cfg);
+    super.postAnalyze(cfg);
+  }
+
+  private void checkCFG(SpecialBlock entry, ControlFlowGraph cfg) {
+
+    Set<Pair<Long, Set<AssignmentNode>>> visited = new HashSet<>();
+    visited.add(Pair.of(entry.getId(), Collections.emptySet()));
+
+    Deque<Pair<BlockImpl, Set<AssignmentNode>>> worklist = new ArrayDeque<>();
+    worklist.add(Pair.of((BlockImpl) entry, Collections.emptySet()));
```

Comment:
Why do we use ids in the `visited` list but `BlockImpl`s in the worklist?

Reply:
Because it's possible to have the same pair<block, set<defs>>, but we should check both of them. Consider the following example:
    void overWrittingVarInLoop() {
        // :: error: missing.alwayscall
        Foo f = new Foo();
        while (true) {
            // :: error: missing.alwayscall
            f = new Foo();
        }
    }
If we use BlockImpl in the visited then it cannot find the second error inside the loop. The reason that I kept BlockImpl in the worklist is that I couldn't retrieve blocks with just blockIds.